### DISCUSSION TITLE
想添加个图片预览的功能，实现方式上存在疑惑

### DISCUSSION THREAD

**Discussion by xuxchao** [vuejs/vitepress#2302] [Q&A]
通过官网了解，发现这种功能想要扩展：
1. 添加一个 Img 组件，在 .md 中写这个组件。但是这种写法会导致在其他 .md 的阅读器上显示不出来，目前是需要能够显示出来的
2. 通过注册一个钩子，监听用户对图片的单击，然后进行预览。感觉这个好像挺麻烦的

不知道有没有办法可以直接改变 .md 的渲染，渲染到是图片的时候可以替换成别的组件这样

**Comment by brc-dd** [comment]
> 不知道有没有办法可以直接改变 .md 的渲染，渲染到是图片的时候可以替换成别的组件这样

This can be done via a custom markdown-it plugin. But it would be better and easier to do (2) instead. There are already existing libraries like `medium-zoom` that do the heavy lifting for you, so you basically need to invoke it in a custom theme. Refer this thread for reference: #854 You'll need to change some of the code as theming has changed since then.

Basically, it would be like this:

```ts
// .vitepress/theme/index.js
import DefaultTheme from 'vitepress/theme'
import Layout from './Layout.vue'

export default { ...DefaultTheme, Layout }
```

```vue
<!--.vitepress/theme/Layout.vue-->
<script setup>
import DefaultTheme from 'vitepress/theme'

// TODO: initialize medium-zoom here on mount
</script>

<template>
  <DefaultTheme.Layout />
</template>
```
